### DISCUSSION TITLE
如何实现像官网主页那样的动态切换主题？

### DISCUSSION THREAD

**Discussion by ByYogi** [ant-design/ant-design#39259] [Q&A]
![image](https://user-images.githubusercontent.com/111735598/205550857-78f2bfa1-0c5a-4cba-afee-2ec8e818601d.png)
我想结合umi实现这种切换器的功能，请教一下大家
![image](https://user-images.githubusercontent.com/111735598/205558462-a37a483f-d591-4722-b9d5-db9a2653c709.png)

**Comment by afc163** [comment]
@MadCcc 文档是不是还不够简单？

**Reply by whaty** [reply]
他就想实现一个官网首页那样的效果！其实我也想，嘻嘻

**Comment by fnoopv** [comment]
官方可以给个实现demo吗 @afc163

**Reply by whaty** [reply]
直接把官网演示用的那个项目开源出来不就看到了

**Comment by BoyYangzai** [comment]
@ByYogi @fnoopv 
我写了一个Demo,可以参考一下😄
https://codesandbox.io/s/antd5-xdong-tai-zhu-ti-demo-eeeniy?file=/demo.tsx

**Reply by fnoopv** [reply]
这样是没问题的，但是真正使用的话ConfigProvider是在最顶层的，主题切换按钮是在navbar的，所以我看其他人是用context来实现的，但是那个哥们是jsx写的，我改成tsx就报错，，，，

**Reply by BoyYangzai** [reply]
> 这样是没问题的，但是真正使用的话ConfigProvider是在最顶层的，主题切换按钮是在navbar的，所以我看其他人是用context来实现的，但是那个哥们是jsx写的，我改成tsx就报错，，，，

@fnoopv 
是的，本质上都是修改 `themeConfig`提供给下面的`Context`，至于如何修改就是由开发者来决定的了

可以看下 AntDesign 官网的实现方式，其实就是在全局 Layout 处加了 ConfigProvider ，点击全局的 `FolatButton - tooltip - 暗黑主题` 的时候，通过`useSearchParams`判断当前是否是dark，如果是dark，就拿到预设好的dark配置，放到了当前的`ConfigProvider`里面
https://github.com/ant-design/ant-design/blob/0eaf2b40cf22d31075eb1083fd9458d10a6ff66b/.dumi/theme/layouts/GlobalLayout.tsx#L22
